google-cloud-compute/src/main/resources/META-INF/native-image/resource-config.json

```diff
@@ -0,0 +1,9 @@
+{
+  "resources":{
+  "includes":[
+    {
+      "pattern":"\\Qdependencies.properties\\E"
```

Comment:
It's supposed to refer to dependencies.properties in gax-2.16.0.jar. It's really interesting that gax is declared as a dependency in the pom but doesn't show up during the native image build phase:

```
 - com.google.api.gax.grpc.nativeimage.GrpcNettyFeature
  - com.google.api.gax.grpc.nativeimage.ProtobufMessageFeature
  - com.google.cloud.nativeimage.features.clients.CloudFunctionsFeature
  - com.google.cloud.nativeimage.features.clients.CloudSqlFeature
  - com.google.cloud.nativeimage.features.clients.SpannerFeature
  - com.oracle.svm.thirdparty.gson.GsonFeature
  - org.graalvm.junit.platform.JUnitPlatformFeature
```
 If we can figure how to make it show up then we can remove this resource-config.json and the reflect-config.json.

Reply:
Memo:

The dependency tree of this branch:

```
[INFO] --- maven-dependency-plugin:3.1.2:tree (default-cli) @ google-cloud-compute ---
[INFO] Verbose not supported since maven-dependency-plugin 3.0
[INFO] com.google.cloud:google-cloud-compute:jar:1.9.1
[INFO] +- com.google.api:api-common:jar:2.1.5:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  \- com.google.auto.value:auto-value-annotations:jar:1.9:runtime
[INFO] +- com.google.api.grpc:proto-google-cloud-compute-v1:jar:1.9.1:compile
[INFO] +- com.google.guava:guava:jar:31.1-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.21.4:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.13.0:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] +- com.google.api:gax:jar:2.16.0:compile
[INFO] |  +- com.google.auth:google-auth-library-credentials:jar:1.6.0:compile
[INFO] |  +- com.google.auth:google-auth-library-oauth2-http:jar:1.6.0:runtime
[INFO] |  \- io.opencensus:opencensus-api:jar:0.31.0:compile
[INFO] |     \- io.grpc:grpc-context:jar:1.45.1:compile
[INFO] +- com.google.api:gax-grpc:jar:2.16.0:compile
[INFO] |  +- io.grpc:grpc-api:jar:1.45.1:compile
[INFO] |  +- io.grpc:grpc-alts:jar:1.45.1:runtime
[INFO] |  |  +- io.grpc:grpc-grpclb:jar:1.45.1:runtime
[INFO] |  |  \- org.conscrypt:conscrypt-openjdk-uber:jar:2.5.1:runtime
[INFO] |  +- io.grpc:grpc-auth:jar:1.45.1:runtime
[INFO] |  +- io.grpc:grpc-netty-shaded:jar:1.45.1:runtime
[INFO] |  |  +- io.perfmark:perfmark-api:jar:0.23.0:runtime
[INFO] |  |  \- io.grpc:grpc-core:jar:1.45.1:runtime
[INFO] |  |     +- com.google.android:annotations:jar:4.1.1.4:runtime
[INFO] |  |     \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.21:runtime
[INFO] |  +- io.grpc:grpc-protobuf:jar:1.45.1:runtime
[INFO] |  |  \- io.grpc:grpc-protobuf-lite:jar:1.45.1:runtime
[INFO] |  +- io.grpc:grpc-stub:jar:1.45.1:runtime
[INFO] |  \- io.grpc:grpc-googleapis:jar:1.45.1:runtime
[INFO] |     \- io.grpc:grpc-xds:jar:1.45.1:runtime
[INFO] |        +- io.grpc:grpc-services:jar:1.45.1:runtime
[INFO] |        +- com.google.re2j:re2j:jar:1.5:runtime
[INFO] |        +- org.bouncycastle:bcpkix-jdk15on:jar:1.67:runtime
[INFO] |        |  \- org.bouncycastle:bcprov-jdk15on:jar:1.67:runtime
[INFO] |        \- io.opencensus:opencensus-proto:jar:0.2.0:runtime
[INFO] +- com.google.api:gax-httpjson:jar:0.101.0:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.9.0:compile
[INFO] |  +- com.google.http-client:google-http-client-gson:jar:1.41.7:runtime
[INFO] |  \- com.google.protobuf:protobuf-java-util:jar:3.19.4:runtime
[INFO] +- org.threeten:threetenbp:jar:1.6.0:compile
[INFO] +- com.google.http-client:google-http-client:jar:1.41.7:compile
[INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |  |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  |  \- commons-codec:commons-codec:jar:1.15:compile
[INFO] |  +- org.apache.httpcomponents:httpcore:jar:4.4.15:compile
[INFO] |  \- io.opencensus:opencensus-contrib-http-util:jar:0.31.0:compile
[INFO] +- com.google.api.grpc:proto-google-common-protos:jar:2.8.3:compile
[INFO] +- com.google.protobuf:protobuf-java:jar:3.19.4:compile
[INFO] +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- com.google.cloud:google-cloud-core:jar:2.6.0:test
[INFO] |  \- com.google.api.grpc:proto-google-iam-v1:jar:1.3.1:test
[INFO] +- com.google.api:gax-httpjson:jar:testlib:0.101.0:test
[INFO] +- com.google.api:gax:jar:testlib:2.16.0:test
[INFO] \- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.244 s
[INFO] Finished at: 2022-04-29T14:08:37-04:00
[INFO] ------------------------------------------------------------------------
```

Feature implementation in gax-2.16.0.jar :

```
~/java-compute/google-cloud-compute $ jar tf ~/.m2/repository/com/google/api/gax/2.16.0/gax-2.16.0.jar |grep Feature
com/google/api/gax/nativeimage/OpenCensusFeature.class
com/google/api/gax/nativeimage/GoogleJsonClientFeature.class
```

I see the latest invocation on this pull request https://source.cloud.google.com/results/invocations/7abbc5cc-93f6-4d5a-832a-e60decfad0ab/log has the two:

```
========================================================================================================================
GraalVM Native Image: Generating 'native-tests'...
========================================================================================================================
[1/7] Initializing...                                                                                   (14.4s @ 0.36GB)
 Version info: 'GraalVM 22.0.0.2 Java 11 CE'
 9 user-provided feature(s)
  - com.google.api.gax.grpc.nativeimage.GrpcNettyFeature
  - com.google.api.gax.grpc.nativeimage.ProtobufMessageFeature
  - com.google.api.gax.nativeimage.GoogleJsonClientFeature
  - com.google.api.gax.nativeimage.OpenCensusFeature
  - com.google.cloud.nativeimage.features.clients.CloudFunctionsFeature
  - com.google.cloud.nativeimage.features.clients.CloudSqlFeature
  - com.google.cloud.nativeimage.features.clients.SpannerFeature
  - com.oracle.svm.thirdparty.gson.GsonFeature
  - org.graalvm.junit.platform.JUnitPlatformFeature
```

Reply:
oh you're right! My bad, I must've missed something in my local setup.

Reply:
Done, the tests are passing without the redundant configurations.